### ISSUE TITLE
Please help me compile for Fedora 24

### ISSUE THREAD

**Issue by David-Else** [phantom-code/airwave#60]
Hello,

I am new to Linux and the instruction for how to compile for Fedora relate to Fedora 20 which is many years old... and I have no idea what is involved :(

It's taken me about 5 hours research to just get to the point of narrowing down this software as the best chance to get Valhalla Vintage Verb working in Linux... it's all I want! (pathetic humble face)

I understand the legal issues with distributing the Steinberg stuff but.... AAARGGGHHHH?! :)

Can anyone help me with a pre-compiled file so I can try to get this working in Fedora 24 and Ardour? There were some ancient versions in this thread, but I would really like the latest:

https://www.kvraudio.com/forum/search.php?keywords=fedora&t=415929&sf=msgonly

The thought of dealing with SELinux issues right now makes me want to vomit!

**Comment by phantom-code** [comment]
Hi!
I don't use RPM based Linux distribution myself thus I can't build the package for you. But... could you please contact the user "StudioDave" on kvr forums and ask him to send you his build? He uses Fedora and I think he will be glad to help you.

**Comment by davephillips** [comment]
We'll get him squared away. :) I'll be happy to send my binaries, but note that I'm on Fedora 21 x86_64. If I recall correctly I was unable to build the 64-bit version of Airwave, so I can send only the 32-bit version now, i.e. he'll only be able to convert 32-bit plugins. I'll double-check to see if I have a 64-bit version from a previous build.

**Comment by Learn-Digital-Audio** [comment]
**Cheers guys!!!**

64bit would be amazing as nearly all my plugins i want to use are 64bit :) Maybe I can locate 32 bit versions if I need to... but 32 bit is legacy now?

I don't yet understand the differences between a Fedora 21 build and a Fedora 24 build, does that mean older versions of dependencies would be compiled into it?

**Comment by davephillips** [comment]
Some time ago I tried building the 64-bit version. Unfortunately the process of installing the required dependencies completely wrecked my rpm and apt systems. In the end I had to re-install the distro, so I am very unwilling to try building the 64-bit version again.

**Comment by davephillips** [comment]
Okay, here's the files from /usr/local/bin in a tarball. Open the tarball in your $HOME directory, it will create a directory named aw4lda. Copy all the files from that directory into your /usr/local/bin, make sure then try running the airwave-manager GUI. Let us know what happens next. :)
[aw4lda.tar.gz](https://github.com/phantom-code/airwave/files/578117/aw4lda.tar.gz)

**Comment by phantom-code** [comment]
If I were use Windows, I would use 64 bit plugins, of course. But in the case of Linux and wine I prefer 32 bit ones. Usually they work better for me.

**Comment by Learn-Digital-Audio** [comment]
I am running the airwave manager right now! :) THANKS!

I think i figured out how to use it,  this Fedora version only for 32bit VST plugins? I tried to add an x64 plugin and Ardour just kept scanning and never accepted it.

I tried to add spitfish.dll, and Ardour did accept it but when I loaded it Ardour just exited and I got the following message. Is this just a plugin that won't work, or maybe something is wrong with my install or the compiled binary? Thanks!

1478611102.856734533          SPITFISH.so : Initializing plugin endpoint 1.3.2
1478611102.856744605          SPITFISH.so : Plugin binary: /home/david/vst-win/SPITFISH.so
1478611102.856760313          SPITFISH.so : WINE prefix:   /home/david/.wine
1478611102.856788115          SPITFISH.so : WINE loader:   /usr/bin/wine32
1478611102.856796126          SPITFISH.so : VST binary:    /home/david/.wine/drive_c/VSTPlugIns/SPITFISH.dll
1478611102.858697646          SPITFISH.so : Host binary:   /usr/local/bin/airwave-host-32.exe
1478611102.858704359          SPITFISH.so : Log level:     trace
1478611102.859054124          SPITFISH.so : Callback thread started
1478611102.859085675          SPITFISH.so : Waiting response from host endpoint...
1478611102.973337838         SPITFISH.dll : Initializing host endpoint 1.3.2
1478611102.975077306         SPITFISH.dll : Waiting for plugin endpoint request...
1478611102.975087633         SPITFISH.dll : Request from plugin endpoint received, sending response
1478611102.975095633         SPITFISH.dll : Initializing VST plugin...
1478611103.270422880         SPITFISH.dll : VST plugin is initialized
1478611103.270435782         SPITFISH.dll : Host endpoint is initialized
1478611103.270463844          SPITFISH.so : Plugin endpoint is initialized
1478611103.270727710         SPITFISH.dll : Audio thread started
1478611103.271314856          SPITFISH.so : Closing plugin
1478611103.271323654          SPITFISH.so : Waiting for callback thread termination...
1478611103.369404359          SPITFISH.so : Callback thread terminated
1478611103.369524314          SPITFISH.so : Waiting for child process termination...
1478611103.372923385         SPITFISH.dll : Control port isn't connected anymore, exiting
1478611103.372935443         SPITFISH.dll : Terminating the host endpoint...
1478611103.372938334         SPITFISH.dll : Waiting for audio thread termination...
1478611103.470908823         SPITFISH.dll : Audio thread terminated
1478611103.481729739         SPITFISH.dll : Host endpoint terminated
1478611103.483594614          SPITFISH.so : Child process terminated
1478611103.483622438          SPITFISH.so : Plugin endpoint terminated
1478611125.839947312          SPITFISH.so : Initializing plugin endpoint 1.3.2
1478611125.839965160          SPITFISH.so : Plugin binary: /home/david/vst-win/SPITFISH.so
1478611125.839992195          SPITFISH.so : WINE prefix:   /home/david/.wine
1478611125.840129155          SPITFISH.so : WINE loader:   /usr/bin/wine32
1478611125.840142343          SPITFISH.so : VST binary:    /home/david/.wine/drive_c/VSTPlugIns/SPITFISH.dll
1478611125.841515127          SPITFISH.so : Host binary:   /usr/local/bin/airwave-host-32.exe
1478611125.841519307          SPITFISH.so : Log level:     trace
1478611125.844284116          SPITFISH.so : Callback thread started
1478611125.844349423          SPITFISH.so : Waiting response from host endpoint...
1478611125.916900362         SPITFISH.dll : Initializing host endpoint 1.3.2
1478611125.918243751         SPITFISH.dll : Waiting for plugin endpoint request...
1478611125.918253915         SPITFISH.dll : Request from plugin endpoint received, sending response
1478611125.918262358         SPITFISH.dll : Initializing VST plugin...
1478611126.021022117         SPITFISH.dll : VST plugin is initialized
1478611126.021034597         SPITFISH.dll : Host endpoint is initialized
1478611126.021070849          SPITFISH.so : Plugin endpoint is initialized
1478611126.021371854         SPITFISH.dll : Audio thread started
1478611126.121466540         SPITFISH.dll : Audio thread terminated
1478611126.132047222         SPITFISH.dll : Audio thread started
1478611126.314094821         SPITFISH.dll : Unable to create window: Resource data not found.

1478611126.314143540         SPITFISH.dll : Terminating the host endpoint...
1478611126.314147634         SPITFISH.dll : Waiting for audio thread termination...
1478611126.332740574         SPITFISH.dll : Audio thread terminated
1478611126.343610129         SPITFISH.dll : Host endpoint terminated
1478611126.344961844          SPITFISH.so : Child process terminated

**Comment by phantom-code** [comment]
> https://github.com/phantom-code/airwave/issues/60
No, this line tells that CreateWindowEx() call failed

**Comment by Learn-Digital-Audio** [comment]
I have successfully added MJUC, TDR Nova and TDR VOS SlickEQ! Congratulations on this software!

Unfortunately 30 seconds into testing when I pressed the ? button in slickEQ I got a crash with this error report:
[backtrace.txt](https://github.com/phantom-code/airwave/files/578399/backtrace.txt)

I think it will take a lot of testing to see what is really working stably. Thanks again davephillips for the upload.

In my humble opinion this plugin could really revolutionize Linux audio and help give the finger to microsoft, but it needs:

- A binary download. 

I can see this is a 'grey area' legally, but there needs to be some 'unofficial' build link. For example, the http://rpmfusion.org guys host all kinds of 'grey area' programs such as mp3 codecs etc. Also there are sites like https://copr.fedorainfracloud.org/ which 

> "is an easy-to-use automatic build system providing a package repository as its output"

- Publicity, which will lead to more testing, which will lead to bug eradication.

I run a site which gets about 200 individual hits a day from computer audio people, but I can't really write a tutorial and review if my readers can't download the software :( I don't mind hosting a compiled version... If Steinberg give me a take down notice I will comply of course, but i really doubt they would bother.

**Comment by draekko** [comment]
@Learn-Digital-Audio  Some plugins require a native dll's to work in wine. Some i've had to add the d2d1.dll by hand to the Wine prefix i was using for them and in one case i had to add native d3dx10  dlls using winetricks. Other plugins might require installing gdiplus also installed using winetricks. For example to get winetricks to use the prefix in a terminal you could try (assuming you have winetricks installed):

env WINEPREFIX=~/.wine.vsts winetricks gdiplus d3dx10 to install both

Other times some will run better with various versions of windows settings and you can tweak that by running:

env WINEPREFIX=~/.wine.vsts winecfg

Each plugin that fails may require special attention to get it running.

Note this is true of any windows based program or app running through Wine. Sometimes though no matter some software will just fail to run in Wine no matter how much tweaking you do usually because some bit of Windows API isn't support or due to bugs in the plugin or software.

**Comment by Learn-Digital-Audio** [comment]
cheers  draekko! if you, or anyone else, has any settings to make valhalla vintage reverb work please share the wealth :)

**Comment by draekko** [comment]
I'm using wine staging just tried it on Ubuntu (using wine staging ppa's) and it worked flawlessly. I piped DSK Dynamic guitar into it. I tried it with the demo from https://valhalladsp.com/shop/reverb/valhalla-vintage-verb/ and am on wine staging 1.9.22. Ran the setup with : 

env WINEPREFIX=~/.wine.vst.valhalla wine ValhallaVintageVerbWinDemo_V1_5_0dot6.exe

it created a new prefix and then i updated with airwave-manager as usual.  Maybe your version of wine is to "old" :)
